### PR TITLE
Tests: Clean migration tests (part 1)

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
During the review of test failure, I noticed our famous false is not true and misleading line on where the assertion happened.
This PR is a clean-up PR. It is modifying the function `confirmVMIPostMigrationAborted` to use `*WithOffset`, removing a couple of redundant functions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/migration_test.go
Comment (EdDev) [inline tests/migration_test.go:356]: Here is an alternative that does not require the reader to interpret this `gstruct` new thing.

```
var migrationState v1.VirtualMachineInstanceMigrationState
if vmi.Status.MigrationState != nil {
    migrationState.Completed = vmi.Status.MigrationState.Completed
    migrationState.AbortStatus = vmi.Status.MigrationState.AbortStatus
}

return migrationState
```

Then equal against `v1.VirtualMachineInstanceMigrationState{Completed: true, AbortStatus: v1.MigrationAbortSucceeded}`

**Review by EdDev — COMMENTED** [review]
Looks nice!
I do not understand how that wrong build option allowed the tests that used it to pass.

I have one suggestion inline.

File: tests/migration_test.go
Comment (xpivarc) [reply tests/migration_test.go:356]: Using equal produces
```
 Timed out after 180.001s.
  Expected
      <v1.VirtualMachineInstanceMigrationState>: {
          StartTimestamp: nil,
          EndTimestamp: nil,
          TargetNodeDomainDetected: false,
          TargetNodeAddress: "",
          TargetDirectMigrationNodePorts: nil,
          TargetNode: "",
          TargetPod: "",
          TargetAttachmentPodUID: "",
          SourceNode: "",
          Completed: true,
          Failed: false,
          AbortRequested: false,
          AbortStatus: "Succeeded",
          MigrationUID: "",
          Mode: "",
          MigrationPolicyName: nil,
          MigrationConfiguration: nil,
          TargetCPUSet: nil,
          TargetNodeTopology: "",
      }
  to equal
      <v1.VirtualMachineInstanceMigrationState>: {
          StartTimestamp: nil,
          EndTimestamp: nil,
          TargetNodeDomainDetected: false,
          TargetNodeAddress: "",
          TargetDirectMigrationNodePorts: nil,
          TargetNode: "",
          TargetPod: "",
          TargetAttachmentPodUID: "",
          SourceNode: "",
          Completed: false,
          Failed: false,
          AbortRequested: false,
          AbortStatus: "Succeeded",
          MigrationUID: "",
          Mode: "",
          MigrationPolicyName: nil,
          MigrationConfiguration: nil,
          TargetCPUSet: nil,
          TargetNodeTopology: "",

```
on the other had `gstruct`

```
Timed out after 180.000s.
  Expected
      <string>: VirtualMachineInstanceMigrationState
  to match fields: {
  .Completed:
  	Expected
  	    <bool>: true
  	to be false
  }
```

I found the latter to be more clear and more valuable. WDYT?

Comment (EdDev) [reply tests/migration_test.go:356]: Yes, it has a better/clearer report but costs more to understand the code.
I leave it up to you, there is no clear answer what is better.

**Review by EdDev — APPROVED** [review]
Thanks!

**Comment by enp0s3** [comment]
Thank you @xpivarc!
/approve

**Comment by xpivarc** [comment]
/hold
Seems we have an issue

File: tests/migration_test.go
Comment (EdDev) [inline tests/migration_test.go:636]: I think you need to use `tests.RunVMIAndExpectLaunchIgnoreWarnings` instead. True for all others as well.

Comment (xpivarc) [reply tests/migration_test.go:636]: Correct to maintain the same logic. But I don't believe we see failures related to this so I am tempted to use this stricter version. WDYT?

Comment (EdDev) [reply tests/migration_test.go:636]: I barely understand why we need to care about warnings in such a dynamic system. But as long as there are no flakes due to this, feel free to keep as is.

**Comment by xpivarc** [comment]
I had to actually take out the fix for kernelboot as it makes the test fails. I verified that everything important is working and found out the source pod is taking too long to terminate. I ran out of time so I will include fix in the next PR.

@iholder-redhat FYI

File: tests/migration_test.go
Comment (xpivarc) [reply tests/migration_test.go]: Done. I think we should default to a version that captures warnings but this can be followed up.

**Review by EdDev — APPROVED** [review]

File: tests/migration_test.go
Comment (iholder-redhat) [inline tests/migration_test.go:688]: Maybe we can also change the literal `240` to some constant?
Perhaps such a constant exists already?

File: tests/migration_test.go
Comment (iholder-redhat) [inline tests/migration_test.go:370]: Maybe the offset can reside in a const too

**Review by iholder-redhat — COMMENTED** [review]
Great job @xpivarc!
Left some minor nit comments

**Comment by iholder-redhat** [comment]
> I had to actually take out the fix for kernelboot as it makes the test fails. I verified that everything important is working and found out the source pod is taking too long to terminate. I ran out of time so I will include fix in the next PR.
> 
> @iholder-redhat FYI

Sure, thanks.
Please just change the PR description to reflect that.

Also, FYI: https://github.com/kubevirt/kubevirt/issues/8191.

File: tests/migration_test.go
Comment (xpivarc) [reply tests/migration_test.go:688]: Not that I am aware of. I have these magic numbers on my todo list. There is plenty of things we can improve :)

**Comment by xpivarc** [comment]
Rebased.

**Review by EdDev — APPROVED** [review]

**Pull request merged**
